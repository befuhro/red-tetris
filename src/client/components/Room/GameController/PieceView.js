import React from 'react';

import Square from '../Square';

const pieceViewStyle = {
    width: '11.95vh',
    height: '11.95vh',
    margin: '2.5vh',
    display: 'flex',
    flexWrap: 'wrap',
    borderColor: '#0f1423',
    borderWidth: '2vh 2vh 2vh 2vh',
    borderRadius: '1vh 1vh 1vh 1vh',
    borderStyle: 'inset'
};

function createArray(piece) {
    let array = Array(16).fill({color: 'white'});
    if (piece !== undefined) {
        piece.position.forEach((position) => {
            array[(position.row * 4) + position.column - 3] = { color: piece.color };
        });
    }
    return (array);
}

function PieceView({ piece }) {
    const array = createArray(piece);

    return (<div style={pieceViewStyle}>
        {
            array.map((square, index) =>
                <Square
                    key={index}
                    index={index}
                    color={square.color}
                />)
        }
    </div>);
}

// PieceView.propTypes = {
//     piece: PropTypes.object.isRequired,
// };

export default PieceView;
