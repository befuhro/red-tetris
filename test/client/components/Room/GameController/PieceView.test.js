import * as React from 'react';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

enzyme.configure({adapter: new Adapter()});

import PieceView from '../../../../../src/client/components/Room/GameController/PieceView';

describe('<Piece>', () => {
    let component;

    beforeAll(() => {
        component = enzyme.mount(<PieceView
            piece={undefined}
        />)
    });
    it('should return PieceView undefined', () => expect(component.html()).toMatchSnapshot());
});
