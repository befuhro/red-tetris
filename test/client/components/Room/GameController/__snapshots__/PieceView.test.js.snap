// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Piece> should return PieceView undefined 1`] = `"<div style=\\"width: 6.8vh; height: 6.8vh; margin: 2.5vh; display: flex; flex-wrap: wrap; border-color: #0f1423; border-width: 2vh 2vh 2vh 2vh; border-radius: 1vh 1vh 1vh 1vh; border-style: inset;\\"><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div></div>"`;
